modules/common/src/main/java/com/ibm/cloud/cloudant/internal/ErrorTransformInterceptor.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * © Copyright IBM Corporation 2024. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.ibm.cloud.cloudant.internal;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.ibm.cloud.sdk.core.util.GsonSingleton;
+
+import okhttp3.Interceptor;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+public class ErrorTransformInterceptor implements Interceptor {
+
+    @Override
+    public Response intercept(Chain chain) throws IOException {
+        // Don't modify the request, but get the response
+        Response response = chain.proceed(chain.request());
+        if (!"HEAD".equals(chain.request().method()) // skip HEAD requests because they have no body
+            && !response.isSuccessful() // skip successful responses
+            && response.body() != null // skip cases with no body
+            && response.body().contentType() != null // skip cases with no content type
+            && "application".equals(response.body().contentType().type())
+            && "json".equals(response.body().contentType().subtype()) // we only want to work with application/json
+        ) {
+            String errorResponse = response.body().string();
+            JsonObject errorBody = null;
+            try {
+                errorBody = GsonSingleton.getGson().fromJson(errorResponse, JsonObject.class);
+                if (errorBody != null) {
+                    // Don't augment if there is already a trace present
+                    if (!errorBody.has("trace")) {
+                        String error = Optional.ofNullable(errorBody.get("error")).map(JsonElement::getAsString).orElse(null);
```

Comment:
A nit, but logically those two lines fit better after `if (!errorBody.has("errors")) {` check

Reply:
updated in [8411bd3](https://github.com/IBM/cloudant-java-sdk/pull/614/commits/8411bd3e6c47d464c766c8b15247cac4ee6bd51b)